spec/amp-email-actions-and-events.md

```diff
@@ -0,0 +1,500 @@
+  <tr>
+    <td><code>methodName</code></td>
+    <td>no</td>
+    <td>This is for elements with default actions.</p><p>This is the method that the target element (referenced by <code>targetId</code>) exposes and you'd like to execute when the event is triggered.</p><p>AMP has a concept of a default action that elements can implement. So when omitting the <code>methodName</code> AMP will execute that default method.</td>
+  </tr>
+  <tr>
+    <td><code>arg=value</code></td>
+    <td>no</td>
+    <td>Some actions, if documented, may accept arguments. The arguments are defined between parentheses in <code>key=value</code> notation. The accepted values are:
+      <ul>
+        <li>simple unquoted strings: <code>simple-value</code></li>
+        <li>quoted strings: <code>"string value"</code> or <code>'string value'</code></li>
+        <li>boolean values: <code>true</code> or <code>false</code></li>
+        <li>numbers: <code>11</code> or <code>1.1</code></li>
+        <li>dot-syntax reference to event data: <code>event.someDataVariableName</code></li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+## Handling multiple events <a name="handling-multiple-events"></a>
+
+You can listen to multiple events on an element by separating the events with a semicolon `;`.
+
+Example: `on="submit-success:lightbox1;submit-error:lightbox2"`
+
+## Multiple actions for one event <a name="multiple-actions-for-one-event"></a>
+
+You can execute multiple actions in sequence for the same event by separating the actions with a comma ','.
+
+Example: `on="tap:target1.actionA,target2.actionB"`
+
+## Globally-defined events and actions <a name="globally-defined-events-and-actions"></a>
+
+AMP defines a `tap` event globally that you can listen to on any HTML element (including AMP elements).
+
+AMP also defines the `hide`, `show` and `toggleVisibility` actions globally that you can trigger on any HTML element.
+
+[tip type="note"]
+
+An element can only be shown if it was previously hidden by a `hide` or `toggleVisibility` action, or by using the [`hidden`](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden) attribute. The `show` action does not support elements hidden by CSS `display:none` or AMP's `layout=nodisplay`.
+
+For example, the following is possible in AMP:
+
+[sourcecode:html]
+<div id="warning-message">Warning...</div>
+
+<button on="tap:warning-message.hide">Cool, thanks!</button>
+[/sourcecode]
+
+
+[/tip]
+
+## Element-specific events <a name="element-specific-events"></a>
+
+### \* - all elements <a name="---all-elements"></a>
+
+<table>
+  <tr>
+    <th>Event</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td><code>tap</code></td>
+    <td>Fired when the element is clicked/tapped.</td>
+  </tr>
+</table>
+
+### Input elements <a name="input-elements"></a>
+
+<table>
+  <tr>
+    <th width="20%">Event</th>
+    <th width="30%">Description</th>
+    <th width="40%">Elements</th>
+    <th>Data</th>
+  </tr>
+  <!-- change -->
```

Comment:
Are these commented lines necessary? It looks like `input` is the only element for which each row is prefaced with one of these comments.

Reply:
I really don't know - they're copy-pasta'd from the OG document. But if it doesn't make sense to you, it doesn't much more sense to me either. lol